app/src/main/java/com/github/ianellis/shifts/location/AndroidLocationRepository.kt

```diff
@@ -27,17 +27,16 @@ class AndroidLocationRepository(
         private const val LOCATION_PERMISSION = Manifest.permission.ACCESS_COARSE_LOCATION
     }
 
-    override fun getLocationAsync(): Deferred<LatLng> {
+    override suspend fun getLocationAsync(): LatLng {
 
-        return GlobalScope.async {
-            if (permissionGranted()) {
+        return if (permissionGranted()) {
                 withContext(Dispatchers.Main) {
```

Comment:
Is there a reason why `getLocationFromManagerAsync` is executed on the main thread?

It can probably be:

```kotlin
override suspend fun getLocationAsync(): LatLng = if (permissionGranted()) {
    getLocationFromManagerAsync()
} else throw PermissionRequiredException(LOCATION_PERMISSION)
```

Reply:
Sadly this does need to be executed on the main thread `locationProvider.lastLocation.addOnCompleteListener` is the line that needs to run on the main thread otherwise we get an exception. This is the design of component from Android. I assume it is this way to enforce adding listeners on one thread